tool/page_tool.c

```diff
@@ -202,24 +193,22 @@ int create_data_file(PageInfo *data, unsigned long data_size, char *path_to_save
     memset(file_buffer, 0, CHUNK_SIZE);
     unsigned long buffer_count = 0;
     unsigned long i = 0;
-    unsigned int flags_data = 0;
-    void *bin_data = NULL;
+
+    uint64_t page_offset = 0;
+    uint32_t flags_data = 0;
     while (i < data_size) {
         flags_data = 0;
-        if (data[i].swapped != 0 || data[i].present != 0) {
-            if (data[i].present != -1) { //unmapped pages have no flags -> flags_data = 0
-                INITIALIZE_CUSTOM_DIRTY(flags_data, data[i].dirty);
-                INITIALIZE_CUSTOM_ANON(flags_data, data[i].anon);
-                INITIALIZE_CUSTOM_PRESENT(flags_data, data[i].present);
+        //unmapped pages have no flags, no reason to include them
+        if (data[i].present != -1 && (data[i].swapped || data[i].present)) {
```

Comment:
Is it possible that both `data[i].swapped` and `data[i].present` == 0?
 I don't understand, why we need that condition.